src/basic/util.c

```diff
@@ -75,9 +77,17 @@ bool in_initrd(void) {
         if (r >= 0)
                 saved_in_initrd = r > 0;
         else
-                saved_in_initrd = access("/etc/initrd-release", F_OK) >= 0 &&
-                                  statfs("/", &s) >= 0 &&
-                                  is_temporary_fs(&s);
+                saved_in_initrd = access("/etc/initrd-release", F_OK) >= 0;
+
+        r = getenv_bool_secure("SYSTEMD_INITRD_IS_RAMFS");
+        if (r < 0 && r != -ENXIO)
+                log_debug_errno(r, "Failed to parse $SYSTEMD_INITRD_IS_RAMFS, ignoring: %m");
+
+        if (r != 0) {
+                saved_in_initrd = saved_in_initrd &&
+                        statfs("/", &s) >= 0 &&
+                        is_temporary_fs(&s);
```

Comment:
while you are at it, please rework to use path_is_temporary_fs(), to simplify things

Reply:
OK, will do.